src/components/common/ImgSlide/ImgSlide.js

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useRef, useState } from 'react';
+
+import * as S from './Img.style';
+
+const ImgSlide = ({ list }) => {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [newImgArr, setNewImgArr] = useState([]);
+  const slideImageBox = useRef();
+
+  const settingArr = [...list];
+
+  const nextIndex = currentIndex === list.length - 1 ? 0 : currentIndex + 1;
+  const prevIndex = currentIndex === 0 ? list.length - 2 : currentIndex - 2;
+
+  useEffect(() => {
+    settingArr.push(list[0]);
+    settingArr.unshift(list[list.length - 1]);
+    setNewImgArr(settingArr);
+  }, []);
+
+  useEffect(() => {
+    const autoMoveImg = setInterval(() => {
+      moveRightSide();
+    }, 3000);
+
+    return () => {
+      clearInterval(autoMoveImg);
+    };
+  }, [newImgArr, currentIndex]);
+
+  const moveRightSide = () => {
+    let changeArr = [...newImgArr];
+    changeArr.shift();
+    changeArr.push(list[nextIndex]);
+    setNewImgArr(changeArr);
+    currentIndex === list.length - 1
+      ? setCurrentIndex(0)
+      : setCurrentIndex(currentIndex + 1);
+    slideImageBox.current.style.transform = `translateX(0%)`;
+    slideImageBox.current.style.transition = `none`;
+
+    setTimeout(() => {
+      slideImageBox.current.style.transform = `translateX(-100%)`;
+      slideImageBox.current.style.transition = `0.5s all ease-in-out`;
+    }, 10);
+  };
+
+  const moveLeftSide = () => {
+    let changeArr = [...newImgArr];
+    changeArr.pop();
+    changeArr.unshift(
+      prevIndex === -1 ? list[list.length - 1] : list[prevIndex]
+    );
+    setNewImgArr(changeArr);
+    currentIndex === 0
+      ? setCurrentIndex(list.length - 1)
+      : setCurrentIndex(currentIndex - 1);
+    slideImageBox.current.style.transform = `translateX(-200%)`;
+    slideImageBox.current.style.transition = `none`;
+
+    setTimeout(() => {
+      slideImageBox.current.style.transform = `translateX(-100%)`;
+      slideImageBox.current.style.transition = `0.5s all ease-in-out`;
+    }, 10);
+  };
+
+  const handleImgSlide = (e) => {
+    const { name } = e.target;
+    name === 'right' ? moveRightSide() : moveLeftSide();
+  };
```

Comment:
함수를 커링하고 동작에 따라 별도의 함수로 분리하면 더 이해하기 쉬운 코드가 될거 같아요
타입스크립트 였다면 direction의 인자로 들어올 수 있는 타입을 "LEFT" | "RIGHT" 유니온으로 제약해서 더 명시적으로 써줄 수 있습니다
```suggestion
  const handleImgSlide = (direction = "RIGHT") => () => {
    direction === 'RIGHT' ? moveRightSide() : moveLeftSide();
  };
  // ...
  onClick={handleImgSlide("RIGHT")}
  onClick={handleImgSlide("LEFT")}
```